src/software/thunderscope/common/proto_configuration_widget.py

```diff
@@ -0,0 +1,306 @@
+
+        """
+
+        # Extract the options from the descriptor, and store it
+        # in the dictionary.
+        options = MessageToDict(
+            descriptor.GetOptions(), preserving_proto_field_name=True
+        )
+
+        try:
+            min_max = options["[TbotsProto.bounds]"]
+        except KeyError:
+            raise KeyError("{} missing ParameterRangeOptions".format(key))
+
+        return {
+            "name": key,
+            "type": "slider",
+            "value": value,
+            "default": value,
+            "limits": (min_max["min_double_value"], min_max["max_double_value"],),
+            "step": 0.01,
+        }
+
+    @staticmethod
+    def __create_enum_parameter(key, value, descriptor):
+        """Converts an enum field in a protobuf to a ListParameter. Uses
+        the options to lookup all possible enum values and provides them
+        as a dropdown option.
+
+        :param key: The name of the parameter
+        :param value: The default value
+        :param descriptor: The proto descriptor
+
+        """
+        options = []
+
+        for enum_desc in descriptor.enum_type.values:
+            options.append(enum_desc.name)
+
+        # The list index is indexed from 1
+        current_enum_index = value - 1
+
+        return parametertree.parameterTypes.ListParameter(
+            name=key,
+            default=None,
+            value=descriptor.enum_type.values[current_enum_index].name,
+            limits=options,
+        )
+
+    @staticmethod
+    def __create_bool_parameter(key, value, descriptor):
+        """Convert a bool field in proto to a BoolParameter
+
+        :param key: The name of the parameter
+        :param value: The default value
+        :param _: The proto descriptor, unused for bool
+
+        """
+        return {"name": key, "type": "bool", "value": value}
+
+    @staticmethod
+    def __create_string_parameter(key, value, descriptor):
+        """Convert a bool field in proto to a BoolParameter
+
+        :param key: The name of the parameter
+        :param value: The default value
+        :param descriptor: The proto descriptor
+
+        """
+        return {"name": key, "type": "bool", "value": value}
+
+    def config_proto_to_param_dict(
+        self,
+        message,
+        search_term=None,
+        current_attr=None,
+        convert_all_fields_to_bools=False,
+    ):
+        """Converts a protobuf to a pyqtgraph parameter tree dictionary
+        that can loaded directly into a ParameterTree
+
+        https://pyqtgraph.readthedocs.io/en/latest/parametertree/index.html
+
+        :param message: The message to convert to a dictionary
+        :param search_term: The search filter
+        :param current_attr: Which attr we are currently on to set
+        :param convert_all_fields_to_bools: 
+                        Sometimes we just want to check/uncheck the fields
+                        of the protobuf (and not actually set an explicit value)
```

Comment:
I'm a bit confused on when we would use `convert_all_fields_to_bools`. 
e.g. what does it mean for `robot_obstacle_inflation_factor` when `convert_all_fields_to_bools` is true?

Reply:
yeah its bit of a weird feature to toss in here, its better for the other class to override the create_* functions and setup this bool feature so I've removed it from here

but its for the arbitrary plotter. we'd want the option to select floats as bools to plot (as opposed to setting the float itself)

Reply:
Im still slightly confused on what you mean by select floats as bools to plot. Does that mean that floats would be converted to 0 or 1 when they're plotted?